src/focus.service.ts

```diff
@@ -616,14 +574,15 @@ export class FocusService {
       }
     }
 
-    const role = el.getAttribute('role');
-    if (role && focusableRoles.indexOf(role) > -1) {
-      return true;
-    }
+    return true;
+  }
 
-    return focusableTags.indexOf(el.tagName) > -1
-      || (!!tabIndex && Number(tabIndex) >= 0)
-      || !!record;
+  /**
+   * Runs a final check, which can be more expensive, run only if we want to
+   * set the element as our next preferred candidate for focus.
+   */
+  private checkFinalFocusable(el: HTMLElement): boolean {
+    return this.isVisible(el);
```

Comment:
In the event that we want to add any other 'final' checks later, having a function call already made for that makes life easier than modifying + extracting two places. JITs will inline this function, so it doesn't have much/any runtime overhead.